File: backend/app/core/digits.py
_PERSIAN = "۰۱۲۳۴۵۶۷۸۹"
_ARABIC = "٠١٢٣٤٥٦٧٨٩"
_LATIN = "0123456789"

_TO_LATIN = {ord(p): l for p, l in zip(_PERSIAN, _LATIN)}
_TO_LATIN.update({ord(a): l for a, l in zip(_ARABIC, _LATIN)})


def to_latin_digits(text: str) -> str:
    return text.translate(_TO_LATIN)


def parse_number(text: str) -> float:
    """Accepts Persian, Arabic-Indic or Latin digits and returns a float."""
    normalised = to_latin_digits(text).strip().replace(",", "")
    return float(normalised)

File: frontend/src/lib/digits.ts
const PERSIAN = "۰۱۲۳۴۵۶۷۸۹";
const ARABIC = "٠١٢٣٤٥٦٧٨٩";

export function toLatinDigits(text: string): string {
  return text.replace(/[۰-۹٠-٩]/g, (ch) => {
    const p = PERSIAN.indexOf(ch);
    if (p !== -1) return String(p);
    const a = ARABIC.indexOf(ch);
    if (a !== -1) return String(a);
    return ch;
  });
}

export function parseNumber(text: string): number {
  return parseFloat(toLatinDigits(text).trim().replace(/,/g, ""));
}

const LATIN_TO_PERSIAN: Record<string, string> = Object.fromEntries(
  "0123456789".split("").map((d, i) => [d, PERSIAN[i]])
);

export function toPersianDigits(value: string | number): string {
  return String(value).replace(/[0-9]/g, (d) => LATIN_TO_PERSIAN[d]);
}
